src/core/service.c

```diff
@@ -1561,21 +1561,47 @@ static int service_spawn(
                 }
         }
 
-        if (flags & EXEC_SETENV_RESULT) {
-                if (asprintf(our_env + n_env++, "SERVICE_RESULT=%s", service_result_to_string(s->result)) < 0)
-                        return -ENOMEM;
+        if ((flags & EXEC_SETENV_RESULT) ^ (flags & EXEC_SETENV_MONITOR_RESULT)) {
+                /* We don't expect both EXEC_SETENV_RESULT and EXEC_SETENV_MONITOR_RESULT to be set. */
+                static const char *monitor_prefix = "";
+                Service *env_source;
 
-                if (s->main_exec_status.pid > 0 &&
-                    dual_timestamp_is_set(&s->main_exec_status.exit_timestamp)) {
-                        if (asprintf(our_env + n_env++, "EXIT_CODE=%s", sigchld_code_to_string(s->main_exec_status.code)) < 0)
-                                return -ENOMEM;
+                if (flags & EXEC_SETENV_MONITOR_RESULT) {
```

Comment:
add empty line above

Reply:
I think we already have one there, see the blank line after Service *env_source;